scio-core/src/main/java/com/spotify/scio/transforms/FutureHandlers.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.spotify.scio.transforms;
+
+import com.google.common.collect.Iterables;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.MoreExecutors;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.function.Function;
+
+/** Utility to abstract away Guava, Java 8 and Scala future handling. */
+public class FutureHandlers {
+
+  /**
+   * Base interface for future handling.
+   * @param <F> future type.
+   * @param <T> value type.
```

Comment:
jw, why is this parameterized as `<T>` ? It seems like `<B>` would be more consistent with the rest of the package unless I'm misreading this!

Reply:
Agree, was just being lazy :sob:
Although `B` is usually for cases like `<A, B, C>`. Call it `V` instead?

Reply:
`V` sounds good! 👍